group-actors/src/main/java/org/sunbird/actors/DeleteGroupActor.java

```diff
@@ -0,0 +1,93 @@
+package org.sunbird.actors;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.actor.core.ActorConfig;
+import org.sunbird.exception.AuthorizationException;
+import org.sunbird.exception.BaseException;
+import org.sunbird.models.MemberResponse;
+import org.sunbird.request.Request;
+import org.sunbird.response.Response;
+import org.sunbird.service.GroupService;
+import org.sunbird.service.GroupServiceImpl;
+import org.sunbird.service.MemberService;
+import org.sunbird.service.MemberServiceImpl;
+import org.sunbird.telemetry.TelemetryEnvKey;
+import org.sunbird.telemetry.util.TelemetryUtil;
+import org.sunbird.util.CacheUtil;
+import org.sunbird.util.GroupRequestHandler;
+import org.sunbird.util.JsonKey;
+import org.sunbird.util.helper.PropertiesCache;
+
+@ActorConfig(
+  tasks = {"deleteGroup"},
+  asyncTasks = {},
+  dispatcher = "group-dispatcher"
+)
+public class DeleteGroupActor extends BaseActor {
+  private CacheUtil cacheUtil = new CacheUtil();
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    String operation = request.getOperation();
+    switch (operation) {
+      case "deleteGroup":
+        deleteGroup(request);
+        break;
+      default:
+        onReceiveUnsupportedMessage("DeleteGroupActor");
+    }
+  }
+
+  /**
+   * This method will delete the group
+   *
+   * @param actorMessage
+   * @throws BaseException
+   */
+  private void deleteGroup(Request actorMessage) throws BaseException {
+    logger.info("DeleteGroup method call");
+    GroupRequestHandler requestHandler = new GroupRequestHandler();
+    String userId = requestHandler.getRequestedBy(actorMessage);
+    String groupId = (String) actorMessage.getRequest().get(JsonKey.GROUP_ID);
+    logger.info("Delete group for the groupId {}", groupId);
+
+    if (StringUtils.isEmpty(userId)) {
+      throw new AuthorizationException.NotAuthorized();
+    }
+    GroupService groupService = new GroupServiceImpl();
+
+    Map<String, Object> dbResGroup = groupService.readGroup(groupId);
+    // Only Group Creator should be able to delete the group
```

Comment:
Can a group creator delete the group if he is only a member, not an ADMIN? Is there a chance that group creator will be made a normal user?

Reply:
As discussed with Mani.. no one can revoke admin privilege from creator. we don't need to check whether creator will not have admin privilege

Reply:
ok